pymeasure/instruments/keithley/keithley2306.py

```diff
@@ -0,0 +1,608 @@
+        """A boolean property that controls whether long integration fast 
+        readings are enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    long_integration_search_enabled = Instrument.control(
+        ":SENS<ch>:LINT:SEAR?", ":SENS<ch>:LINT:SEAR %d",
+        """A boolean property that controls whether long integration search 
+        is enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    long_integration_detect_enabled = Instrument.control(
+        ":SENS<ch>:LINT:DET?", ":SENS<ch>:LINT:DET %d",
+        """A boolean property that controls whether long integration detection 
+        mode is enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    source_voltage = Instrument.control(
+        ":SOUR<ch>:VOLT?", ":SOUR<ch>:VOLT %g",
+        """A floating point property that controls the source voltage in volts, 
+        which takes values between 0 and 15. """,
+        validator=truncated_range,
+        values=[0, 15],
+    )
+
+    source_voltage_protection = Instrument.control(
+        ":SOUR<ch>:VOLT:PROT?", ":SOUR<ch>:VOLT:PROT %g",
+        """A floating point property that controls the source voltage protection 
+        offset in volts, which takes values between 0 and 8. """,
+        validator=truncated_range,
+        values=[0, 8],
+    )
+
+    source_voltage_protection_enabled = Instrument.measurement(
+        ":SOUR<ch>:VOLT:PROT:STAT?",
+        """A boolean property that returns the source voltage protection state. """
+    )
+
+    source_voltage_protection_clamp_enabled = Instrument.control(
+        ":SOUR<ch>:VOLT:PROT:CLAM?", ":SOUR<ch>:VOLT:PROT:CLAM %d",
+        """A boolean property that controls whether source voltage protection 
+        clamp is enabled. """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    source_current_limit = Instrument.control(
+        ":SOUR<ch>:CURR?", ":SOUR<ch>:CURR %g",
+        """A floating point property that controls the source current limit in 
+        amps, which takes values between 0.006 and 5. """,
+        validator=truncated_range,
+        values=[0.006, 5],
+    )
+
+    source_current_limit_type = Instrument.control(
+        ":SOUR<ch>:CURR:TYPE?", ":SOUR<ch>:CURR:TYPE %s",
+        """A string property that controls source current limit type, which can 
+        take the values 'limit' or 'trip'. """,
+        validator=strict_discrete_set,
+        values={'limit': 'LIM', 'trip': 'TRIP'},
+        map_values=True,
+    )
+
+    source_current_limit_enabled = Instrument.measurement(
+        ":SOUR<ch>:CURR:STAT?",
+        """A boolean property that returns the source current limit state. """
+    )
+
+    read = Instrument.measurement(
```

Comment:
Perhaps a better name for this property would be `reading`. There is another SCPI command that returns an array, and maybe that could be `readings`.

Reply:
also `Instrument.read()` is already an pretty central method, and `read` sounds more like a verb to me, and properties are often/typically nouns/adjectives (in the spirit of command-query-separation).
(drive by comment to the comment, it will take time until I get around for a proper review.)

Reply:
I took a deeper look at the SCPI API and there are roughly 3 modes: 

1. Trigger and return a measurement based on the previously set sense mode (voltage, current, etc.), 
2. Trigger and return a measurement while also specifying the sense mode (which sets the sense mode as a side-effect)
3. Just fetch existing measurements (without triggering, or specifying sense mode)

Additionally, you can request either a single value or an array (where the number of values returned is tied to the number of values used for averaging).

So, given all of the above, I added a whole slew of properties allowing the user to obtain data in a way that fairly closely adheres to the way the SCPI API works.